authentication/email_validation_api.py

```diff
@@ -0,0 +1,232 @@
+        """Return true if the size of the file is at or past the limit"""
+        return self.file.tell() > MAX_CSV_FILE_SIZE
+
+
+@contextmanager
+def csv_batch_file(name: str):
+    """
+    Get a temporary file and CSV writer
+
+    Args:
+        name(str): the name of the operation/upload
+
+    Returns:
+        CsvBatchFile: the batch and associated state and files
+    """
+    with tempfile.NamedTemporaryFile(
+        prefix=name, suffix=".csv", mode="w", delete=False
+    ) as file:
+        writer = DictWriter(file, fieldnames=["email"])
+        writer.writeheader()
+
+        yield CsvBatchFile(name, file, writer)
+
+
+def get_api_key():
+    """Return the Mailgun API key from the Anymail settings"""
+    return get_anymail_setting("API_KEY", esp_name="MAILGUN")
+
+
+def send_to_mailgun(csv_batch):
+    """
+    Send the CSV file to mailgun
+
+    """
+    with open(csv_batch.file.name, "r") as csv_file:
+        # send the temporary file up to mailgun
+        resp = requests.post(
+            urljoin(VALIDATION_API_URL, csv_batch.name),
+            auth=("api", get_api_key()),
+            files={"file": ("file", csv_file.read())},
+            timeout=60 * 5,
+        )
+        resp.raise_for_status()
+
+
+def batch_validation_csv_files(
+    users,
+    list_name_base: str,
+    *,
+    db_chunk_size: int = 1000,
+    upload_batch_size: int = 5000,
+):
+    """
+    Yields CSV files to upload to mailgun
+    """
+
+    # create the iterator here so it can be reused across while loop steps
+    user_chunks_iter = chunks(users.only("email"), chunk_size=db_chunk_size)
+
+    batch_idx = 1
+    while True:
+        with csv_batch_file(f"{list_name_base}-{batch_idx}") as csv_batch:
+            # this transaction will mainly get rolled back if there's an error within the yield
+            with transaction.atomic():
+                batch_size = 0
+                for chunk in user_chunks_iter:
+                    csv_batch.writer.writerows(
+                        [{"email": user.email} for user in chunk]
+                    )
+
+                    EmailValidation.objects.bulk_create(
+                        [
+                            EmailValidation(
+                                user=user, email=user.email, list_name=csv_batch.name
+                            )
+                            for user in chunk
+                        ]
+                    )
+
+                    batch_size += len(chunk)
+
+                    if csv_batch.is_full() or batch_size >= upload_batch_size:
+                        break
+
+                if not csv_batch.has_data():
+                    return
+
+                # ensure everything is flushed to disk before we yield
+                csv_batch.file.flush()
+
+                yield csv_batch
+
+        batch_idx += 1
+
+
+def start_user_email_validation(
+    users,
+    list_name_base: str,
+    *,
+    db_chunk_size: int = 1000,
+    upload_batch_size: int = 20000,
```

Comment:
the default batch size is different in different functions. Can you remove the default everywhere except the management command?